### PR TITLE
perf: move rvalue scalars to variant

### PR DESCRIPTION
**Pull request by lukasberbuer**
Avoid copies of rvalue scalars using `Variant::setScalarCopy` (if possible).

As discussed in #174 with @lbckmnn.

### CODE REVIEW

File: include/open62541pp/types/Variant.h [hunk 1]
```diff
@@ -457,9 +460,15 @@ void Variant::setArrayImpl(
...
-    auto native = detail::allocateUniquePtr<T>(dataType);
-    *native = detail::copy(value, dataType);
+void Variant::setScalarCopyImpl(T&& value, const UA_DataType& dataType) {
+    using ValueType = std::remove_cv_t<std::remove_reference_t<T>>;
+    auto native = detail::allocateUniquePtr<ValueType>(dataType);
+    // https://stackoverflow.com/a/50086839/9967707
+    if constexpr (std::is_lvalue_reference_v<T>) {
+        *native = detail::copy(value, dataType);
+    } else {
+        *native = std::exchange(value, {});
```
Comment (lbckmnn) [inline include/open62541pp/types/Variant.h:470]: can't we just move assign here?

File: include/open62541pp/types/Variant.h [hunk 2]
```diff
@@ -507,11 +516,21 @@ struct VariantHandler<VariantPolicy::Copy> {
         var.setScalarCopy(value);
     }
 
+    template <typename T, typename = std::enable_if_t<!std::is_reference_v<T>>>
```
Comment (lbckmnn) [inline include/open62541pp/types/Variant.h:519]: yeah, it is really hard to see why this `std::enable_if` is needed, just by looking at the code.

**Review by lbckmnn — COMMENTED** [review]
Nice idea, but I am not sure if it is worth the extra complexity.

File: include/open62541pp/types/Variant.h (hunk 1 above)
Comment (lukasberbuer) [reply include/open62541pp/types/Variant.h:470]: Integral values and simple C `struct`s will just copy if using `std::move`. Zero-initializing the source object prevents maybe a necessary deallocation of the source object.

File: include/open62541pp/types/Variant.h (hunk 2 above)
Comment (lukasberbuer) [reply include/open62541pp/types/Variant.h:519]: Totally agree.. let leave it as a draft PR.

**Pull request closed without merging**